src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -457,7 +457,7 @@ internal struct ZipCentralDirectoryFileHeader
         public ushort FileCommentLength;
         public int DiskNumberStart;
         public ushort InternalFileAttributes;
-        public uint ExternalFileAttributes;
+        public int ExternalFileAttributes;
```

Comment:
Why the change from `uint` to `int` here?

In general, since this doesn't represent a "signed" number, it is better to use `uint` - just like all the other fields in this struct. This seems pretty inconsistent now.

Reply:
The `public ZipArchiveEntry.ExternalAttributes` is an `int`, and I changed this to get rid of the casts.
I can change it back (and add back the casts) if you prefer?

Reply:
It seems so inconsistent now with the rest of the code and feels like an unnecessary change.

if

```C#
internal const int DefaultFileExternalAttributes =
```

was declared as a `uint`, I think there would be a lot less casting happening in your new code.